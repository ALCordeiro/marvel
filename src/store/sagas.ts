import { all } from 'redux-saga/effects';
import { CharactersSaga } from './ducks/characters/sagas';

export function* rootSagas() {
    yield all([...CharactersSaga]);
}
