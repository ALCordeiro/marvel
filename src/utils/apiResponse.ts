export const characters: any = [
    {
        comics: {
            available: 12,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/comics',
            items: Array(12),
            returned: 12,
        },
        description: '',
        events: {
            available: 1,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/events',
            items: Array(1),
            returned: 1,
        },
        id: 1011334,
        modified: '2014-04-29T14:18:17-0400',
        name: '3-D Man',
        resourceURI: 'http://gateway.marvel.com/v1/public/characters/1011334',
        series: {
            available: 3,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/series',
            items: Array(3),
            returned: 3,
        },
        stories: {
            available: 21,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/stories',
            items: Array(20),
            returned: 20,
        },
        thumbnail: {
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
            extension: 'jpg',
        },
        map: () => 'test',
        length: () => 1,
        filter: () => 'test',
    },
];

export const character: any = {
    state: {
        id: 10,
        name: 'Rubick',
        description: 'The Grand magus',
        comics: {
            available: 1,
            items: [
                {
                    name: 'The Granddd magus',
                },
            ],
        },
        series: {
            available: 1,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1017100/series',
            items: [
                {
                    name: 'FREE COMIC BOOK DAY 2013 1 (2013)',
                    resourceURI:
                        'http://gateway.marvel.com/v1/public/series/17765',
                },
            ],
            returned: 1,
        },
        thumbnail: {
            extension: 'jpg',
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16',
        },
    },
};
