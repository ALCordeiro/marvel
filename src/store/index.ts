import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { CharactersState } from './ducks/characters/types';
import rootReducer from './ducks/rootReducer';
import { rootSagas } from './sagas';

export interface ApplicationState {
    characters: CharactersState;
}

const middleware = [];
const initialState = {};
const sagaMiddleware = createSagaMiddleware();
middleware.push(sagaMiddleware);

const store = createStore(
    rootReducer,
    initialState,
    composeWithDevTools(applyMiddleware(...middleware)),
);

sagaMiddleware.run(rootSagas);

export default store;
