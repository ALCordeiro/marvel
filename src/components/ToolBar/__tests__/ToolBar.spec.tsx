/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../store';
import ToolBarComponent from '../ToolBar';
import { characters } from '../../../utils/apiResponse';

let wrapper;

beforeEach(() => {
    wrapper = render(
        <Provider store={store}>
            <Router>
                <ToolBarComponent characters={characters} />
            </Router>
        </Provider>,
    );
});

test('should render', async () => {
    const { container } = wrapper;
    expect(container.firstChild);
});

test('should match snapshot', () => {
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
});
