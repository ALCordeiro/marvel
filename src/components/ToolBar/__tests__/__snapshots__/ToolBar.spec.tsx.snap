// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="sc-bdnxRM fdHRjc"
>
  <div
    class="sc-gtsrHT jTOMQt"
  >
    <p>
      Encontrado(s) 
      1
       her√≥i(s)
    </p>
  </div>
  <div
    class="sc-dlnjwi iUsAoZ"
  >
    <div
      class="sc-hKFxyN bIItPl"
    >
      <img
        alt="marvel-super-hero"
        src="http://localhost/assets/super-hero.svg"
      />
      <p>
        Ordenar por nome - A/Z
      </p>
      <label
        class="sc-eCApnc diXFVk"
        for="sorter"
      >
        <input
          id="sorter"
          type="checkbox"
        />
        <span
          class="slider round"
        />
      </label>
    </div>
    <p>
      test
    </p>
  </div>
</div>
`;
